### DISCUSSION TITLE
[2023-11-06] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#8232] [Weekly Demo Day]
- James: Change widget for Order By. #8226
- Radosław: Implement `Table.lookup_and_replace` in Database #8146
- Radosław: Improve performance of `Join_Condition.Between` by sorting on one dimension 
  #8212
  - Also includes #8204
- Radosław: Simplify internal logic of `Table.order_by`, avoid unnecessary warning 
  #8221
- Radosław: finishing a PR from last week (review, fixing tests)
  #8159
  - Enabled some assertions that used to be just comments in the library code, thanks to the `Runtime.assert` implemented by Pavel.
- Greg: Overhaul the random number and item generation code #8127
- Greg: Implement Table.format similar to Table.parse allowing to format columns in bulk #8150
- Greg: Constant columns (in expressions and Column_Operations) should have clearer names #8188
- Greg: remove extra content/home -- do not review, just testing #8208
- Greg: Allow the creation of a constant column on an in-memory table with no rows. #8218
- Jaroslav: Eliminate references to Truffle nodes & co. in the compiler #8172
- Jaroslav: Caches belong to runtime, not IR compiler #8178
- Jaroslav: Removal of useless ApplicationSaturation phase #8181
- Jaroslav: Allow reassigning of BindingsMap #8190
- Jaroslav: Introducing engine/runtime-compiler project #8197
- Jaroslav: Organizing Enso projects into physical folders #8205
- Jaroslav: Store IR caches in suitable format to read them "on demand" #8207
- Hubert: Apply custom log levels to Truffle logger #8162
- Hubert: Use TruffleLogger bound to the engine to prevent illegal usage #8169
- Hubert: Upgrade directory-watcher library #8201
- Hubert: OpenFileCmd sends a reply when finished #8225
- Pavel: Add disable-private-check option #8202
- Pavel: Upgrade Enso to GraalVM for jdk 21 #7991
- Dmitry: Inline Execution #8148
- Dmitry: Unlock resources initialization asynchronously #8206

**Comment by radeusgd** [comment]
I've added a few new benchmarks to be able to compare the Join performance on the `Between` conditions.

![image](https://github.com/enso-org/enso/assets/1436948/2fa64ce6-1dba-43db-93f0-a2901205db54)
